code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/dao/impl/AuthDaoImpl.java

```diff
@@ -129,70 +161,247 @@ public void updateAccount(Account account, String name, String kind, String exte
+    }
+
+    @Override
+    public boolean hasAccessToProject (long projectId, Long usingAccount, boolean isAdmin, Set<ExternalId> externalIds) {
+        return isProjectMember(projectId, usingAccount, isAdmin, externalIds);
+    }
+
+    @Override
+    public boolean isProjectOwner (long projectId, Long usingAccount, boolean isAdmin, Set<ExternalId> externalIds){
+        if (externalIds == null) {
+            return false;
+        }
+        if (isAdmin) {
+            return true;
+        }
+        if (usingAccount !=  null && usingAccount.equals(projectId)){
+            return false;
+        }
+        Set<ProjectMemberRecord> projectMembers = new HashSet<>();
+        Condition allMembers = DSL.falseCondition();
+        for (ExternalId id: externalIds){
+            allMembers = allMembers.or(PROJECT_MEMBER.EXTERNAL_ID.eq(id.getId())
+                    .and(PROJECT_MEMBER.EXTERNAL_ID_TYPE.eq(id.getType()))
+                    .and(PROJECT_MEMBER.ROLE.eq(ProjectConstants.OWNER))
+                    .and(PROJECT_MEMBER.PROJECT_ID.eq(projectId))
+                    .and(PROJECT_MEMBER.STATE.eq(CommonStatesConstants.ACTIVE))
+                    .and(PROJECT_MEMBER.REMOVED.isNull()));
+        }
+        projectMembers.addAll(create().selectFrom(PROJECT_MEMBER).where(allMembers).fetch());
+        return !projectMembers.isEmpty();
     }
     
-    public ObjectManager getObjectManager() {
-        return objectManager;
+    public boolean isProjectMember (long projectId, Long usingAccount, boolean isAdmin,Set<ExternalId> externalIds){
+        if (externalIds == null) {
+            return false;
+        }
+        if ((usingAccount != null && usingAccount.equals(projectId)) || isAdmin) {
+            return true;
+        }
+        Set<String> roles =  new HashSet<>();
+        roles.add(ProjectConstants.OWNER);
+        roles.add(ProjectConstants.MEMBER);
+        Set<ProjectMemberRecord> projectMembers = new HashSet<>();
+        Condition allMembers = DSL.falseCondition();
+        for (ExternalId id: externalIds){
+            allMembers = allMembers.or(PROJECT_MEMBER.EXTERNAL_ID.eq(id.getId())
+                    .and(PROJECT_MEMBER.EXTERNAL_ID_TYPE.eq(id.getType()))
+                    .and(PROJECT_MEMBER.ROLE.in(roles))
+                    .and(PROJECT_MEMBER.PROJECT_ID.eq(projectId))
+                    .and(PROJECT_MEMBER.STATE.eq(CommonStatesConstants.ACTIVE))
+                    .and(PROJECT_MEMBER.REMOVED.isNull()));
+        }
+        projectMembers.addAll(create().selectFrom(PROJECT_MEMBER).where(allMembers).fetch());
+        return !projectMembers.isEmpty();
     }
 
-    @Inject
-    public void setObjectManager(ObjectManager objectManager) {
-        this.objectManager = objectManager;
+    @Override
+    public Account updateProject(Account project, String name, String description) {
+        Map<TableField<AccountRecord, String>, String> properties = new HashMap<>();
```

Comment:
Just was reusing code that had already been there from previous implementation. I can change if you'd like.